data/contract_docs/logs.md

```diff
@@ -0,0 +1,97 @@
+# Logs
+
+Status: EARLY Draft/Proposal
+
+## LogLines
+
+### Properties and field requirements
+
+- **Time field** - _required_
+  - There must be at least one non nullable time field
+  - If there are multiple time fields present, following will decide the priority
+    - First matching time field with name `timestamp`
+    - or first matching time field with name `ts`
+    - or first matching time field
+- **Message field** - _required_
+  - There must be at lease one non nullable string field must present
+  - If more than one string fields found, the following will decide the priority
+    - First matching string field with name `body`
+    - or first matching string field with name `message`
+    - or first matching string field
+- **Severity field** - _optional_
+  - This is optional field
+  - Level/Severity of the log line can be represented with this field.
+  - This have to be a string field. (either nullable or non-nullable string field)
+  - Log field will be decided in the following order
+    - First matching string field with name `severity`
+    - or first matching string field with name `level`
+  - If no level field found, consumers/client will decide the log level. Example: logs panels will try to parse the message field and determine the log level
+  - Log level can be one of the values specified in the docs [here](https://grafana.com/docs/grafana/latest/explore/logs-integration/)
```

Comment:
(writing this down so i don't forget it :) )

one complication here is that systems like Elastic or Loki store this as a string, so it can have any value. question is, how do we map levels that are not in the "allowed" enum.
maybe we make them "unknown"?